muffnn/mlp/base.py

```diff
@@ -156,30 +157,32 @@ def partial_fit(self, X, y, monitor=None, **kwargs):
 
         # Train the model with the given data.
         with self.graph_.as_default():
-            n_examples = X.shape[0]
-            indices = np.arange(n_examples)
-
-            for epoch in range(self.n_epochs):
-                self._random_state.shuffle(indices)
-                for start_idx in range(0, n_examples, self.batch_size):
-                    batch_ind = indices[start_idx:start_idx + self.batch_size]
-                    feed_dict = self._make_feed_dict(X[batch_ind],
-                                                     y[batch_ind])
-                    obj_val, _ = self._session.run(
-                        [self._obj_func, self._train_step],
-                        feed_dict=feed_dict)
-                    _LOGGER.debug("objective: %.4f, epoch: %d, idx: %d",
-                                  obj_val, epoch, start_idx)
-
-                _LOGGER.info("objective: %.4f, epoch: %d, idx: %d",
-                             obj_val, epoch, start_idx)
-
-                if monitor:
-                    stop_early = monitor(epoch, self, {'loss': obj_val})
-                    if stop_early:
-                        _LOGGER.info(
-                            "stopping early due to monitor function.")
-                        return self
+            # Initialize dataset iterator
+            # TODO: should this be in the with clause above
+
+            self._session.run(self.iterator.initializer,
+                              feed_dict={self._features_placeholder: X,
+                                         self._labels_placeholder: y})
+
+            while True:
+                try:
+                    # TODO: add a global step variable
+                    # Taining step
+                    obj_val, _ = self._session.run([self._obj_func, self._train_step])
+
+                    # TODO: reduce the number of saved values, this should be replaced with a summary writer
+                    _LOGGER.debug("objective: %.4f, epoch: %d, idx: %d", obj_val)
+
+                    # TODO: figure out how to catch end of epoch to keep this
```

Comment:
The number of training steps per epoch can be precomputed. If this does not work, we can just run one epoch at a time even when the user asks for more than one. This will still eliminate overhead but would allow for us to catch the end of each one.